chandra_aca/dark_subtract.py

```diff
@@ -0,0 +1,207 @@
+    """
+    Get the scaled dark current values for a table of ACA images.
+
+    This scales the dark current to the appropriate temperature and integration time,
+    returning the dark current in DN matching the ACA images in img_table.
+
+    Parameters
+    ----------
+    img_table : astropy.table.Table
+        Table of ACA images with columns 'IMG', 'IMGROW0_8X8', 'IMGCOL0_8X8', 'INTEG'.
+    img_dark : 1024x1024 array
+        Dark calibration image.
+    tccd_dark : float
+        Reference temperature of the dark calibration image.
+    t_ccds : array
+        Cheta temperature values at the times of img_table.
+
+    Returns
+    -------
+    imgs_dark : np.array (len(img_table), 8, 8)
+        Temperature scaled dark current for each ACA image in img_table (DN).
+
+    """
+    if len(img_table) != len(t_ccds):
+        raise ValueError("img_table and t_ccds must have the same length")
+
+    if img_dark.shape != (1024, 1024):
+        raise ValueError("dark image shape is not 1024x1024")
+
+    imgs_dark_unscaled = get_dark_backgrounds(
+        img_dark,
+        img_table["IMGROW0_8X8"],
+        img_table["IMGCOL0_8X8"],
+    )
+
+    imgs_dark = np.zeros((len(img_table), 8, 8), dtype=np.float64)
+
+    # Scale the dark current at each dark cal 8x8 image to the ccd temperature and e-/s
+    for i, (img_dark_unscaled, t_ccd_i, img) in enumerate(
+        zip(imgs_dark_unscaled, t_ccds, img_table, strict=True)
+    ):
+        img_scaled = dark_temp_scale_img(img_dark_unscaled, tccd_dark, t_ccd_i)
+        # Default to using 1.696 integ time if not present in the image table
+        integ = img["INTEG"] if "INTEG" in img.colnames else 1.696
+        imgs_dark[i] = img_scaled * integ / 5
+
+    return imgs_dark
+
+
+def get_dark_backgrounds(raw_dark_img, imgrow0, imgcol0, size=8):
+    """
+    Get dark background cutouts at a set of ACA image positions.
+
+    Parameters
+    ----------
+    raw_dark_img : np.array
+        Dark calibration image.
+    imgrow0 : np.array (int)
+        Row of ACA image.
+    imgcol0 : np.array (int)
+        Column of ACA image.
+    size : int, optional
+        Size of ACA image (default=8).
+
+    Returns
+    -------
+    imgs_dark : np.array (len(imgrow0), size, size)
+        Dark backgrounds for image locations sampled from raw_dark_img (e-/s).
+        Pixels outside raw_dark_img are set to 0.0.
+    """
+    CCD_MAX = 1024
+    CCD_MIN = 0
+
+    @numba.jit(nopython=True)
+    def staggered_aca_slice(array_in, array_out, row, col):
+        for idx in np.arange(len(row)):
+            row_out = np.zeros((size, size), dtype=np.float64)
+            for i, r in enumerate(np.arange(row[idx], row[idx] + size)):
+                for j, c in enumerate(np.arange(col[idx], col[idx] + size)):
+                    if r >= CCD_MIN and r < CCD_MAX and c >= CCD_MIN and c < CCD_MAX:
+                        row_out[i, j] = array_in[r, c]
```

Comment:
I'm sure there are nicer and faster ways to get the values right up to the dark map edges, but this was easiest for me to read and since we're numba-ing it anyway...